osfoffline/database/models.py

```diff
@@ -81,6 +105,14 @@ def validate_top_level(self, key, top_level):
             assert self.parent is not None
         return top_level
 
+    @validates('sync')
+    def validate_sync(self, key, sync):
+        if not self.parent_id:
+            assert sync is True
```

Comment:
I chose to follow the existing convention used in other validators.

This validator is in place largely to improve feedback to developers who might make future assumptions about the meaning of Node.sync.